### ISSUE TITLE
extractLinkHeader not always extracting the desired link

### ISSUE THREAD

**Issue by Carles90** [ohmybrew/Basic-Shopify-API#45]
Hi,

I found very useful the new functionality about parsing the "Link" header to the "link" property, for pagination purposes.
This functionality works perfect for the first page, because we have the "page_info" string in the "link" property.
The problem comes in the second page, where the "Link" header returns two links:

```
<https://xxxx.myshopify.com/admin/api/2019-10/products.json?limit=3&page_info=eyJkaXJlY3Rpb24iOiJwcmV2IiwibGFzdF9pZCI6MjM5ODQxNjg5NjA3NywibGFzdF92YWx1ZSI6IlNpbGxhIGRlIHJ1ZWRhcyBhIHBlZGFsZXMifQ>; rel="previous", 
<https://xxxx.myshopify.com/admin/api/2019-10/products.json?limit=3&page_info=eyJkaXJlY3Rpb24iOiJuZXh0IiwibGFzdF9pZCI6MjQ2NzI4MzEwNzkxNywibGFzdF92YWx1ZSI6IlRlc3Qgbm90IHB1Ymxpc2hlZCJ9>; rel="next"
```

The regex in the "extractHeaderLink" will always take the page_info from the first link, and forget about the second one.
This is not good if you are expecting the "link" parameter to loop over a big collection in Shopify API, because you are getting in an infinite loop in page 1, page 2, page 1, page 2. This is because the "previous" page link, always appear in first place in the header (except in page 1, where there is only a "next" page link).

My proposal, is to have both "previous" and "next" properties in BasicShopifyAPI class, along with the "link", which takes the correct "page_info" link for the previous page and for the next one respectively.

Thanks!

**Comment by ohmybrew** [comment]
Ah overlooked this... result of being sick and trying to code haha
Adjusting!

**Comment by ohmybrew** [comment]
Solved in 6.1.1 ... there is now ->next and ->previous... which default to null if non existant.

**Comment by Carles90** [comment]
I confirm this is working now for me.

Thanks for your fast fix! 😉
